### PR TITLE
[508] Make part of the Sirius debugging view API

### PR DESCRIPTION
**Pull request by florianbarbin**
* Create a new feature org.eclipse.sirius.debug.feature
* This feature will reference a new plugin org.eclipse.sirius.debug that expose the AbstractDebugView and the TabularReport classes

Bug: https://github.com/eclipse-sirius/sirius-desktop/issues/508

### CODE REVIEW

File: plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java [hunk 1]
```diff
@@ -145,12 +135,12 @@ public void selectionChanged(IWorkbenchPart part, ISelection selection) {
     /**
      * Get the main object selected from the Eclipse ISelection.
      */
-    private Option<Object> getSelectedElement(ISelection selection) {
+    private Optional<Object> getSelectedElement(ISelection selection) {
         if (selection instanceof IStructuredSelection) {
```
Comment (lredor) [inline plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java:139]: You can directly use the instanceof cast:
```suggestion
        if (newSelection instanceof IStructuredSelection iss) {
```

File: plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java [hunk 2]
```diff
@@ -145,12 +135,12 @@ public void selectionChanged(IWorkbenchPart part, ISelection selection) {
...
             IStructuredSelection iss = (IStructuredSelection) selection;
-            return Options.newSome(iss.getFirstElement());
+            return Optional.of(iss.getFirstElement());
         }
-        return Options.newNone();
+        return Optional.empty();
     }
```
Comment (lredor) [inline plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java:145]: CheckStyle rule: Add javadoc for `getAdapterFactory` method
```
    /**
     * Return the adapter factory specific for the given view.
     * 
     * @return the adapter factory specific for the given view.
     */
    protected AdapterFactory getAdapterFactory() { 
```

File: plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java [hunk 3]
```diff
@@ -77,7 +74,7 @@ public Action(String label, String description, Runnable body) {
     /**
      * The currently selection object.
      */
-    protected Object selection;
+    public Object selection;
 
     @Override
     public void createPartControl(Composite parent) {
```
Comment (lredor) [inline plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java:80]: Checkstyle rule: for the above line checkstyle must be disabled:
```
        // CHECKSTYLE:OFF
        info = new Text(parent, SWT.MULTI | SWT.READ_ONLY | SWT.WRAP | SWT.BORDER | SWT.H_SCROLL | SWT.V_SCROLL);
        // CHECKSTYLE:ON
```

File: plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/TabularReport.java [hunk 4]
```diff
@@ -10,25 +10,26 @@
...
  * 
  * @author pcdavid
  */
+@SuppressWarnings("nls")
 public class TabularReport {
     private final List<String> headers;
 
     private final List<List<String>> lines;
 
     public TabularReport(String... headers) {
```
Comment (lredor) [inline plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/TabularReport.java:31]:
```suggestion
    /**
     * Default constructor.
     * 
     * @param headers
     *            List of header names.
     */
    public TabularReport(String... headers) {
```

File: plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/TabularReport.java [hunk 5]
```diff
@@ -37,7 +38,7 @@ public void addLine(List<String> data) {
     }
 
     public void addLine(String... data) {
```
Comment (lredor) [inline plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/TabularReport.java:40]: Checkstyle rule: method to comment (use same comment for above method)
```suggestion
    /**
     * Add one line for each {@link String} of data list.
     * 
     * @param data
     *            The list of lines to add.
     */
    public void addLine(String... data) {
```

File: plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/TabularReport.java [hunk 6]
```diff
@@ -37,7 +38,7 @@ public void addLine(List<String> data) {
...
-        addLine(new ArrayList<String>(Arrays.asList(data)));
+        addLine(new ArrayList<>(Arrays.asList(data)));
     }
 
     public String print() {
```
Comment (lredor) [inline plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/TabularReport.java:44]:
```suggestion
    /**
     * Return a string representation of the {@link TabularReport}.
     *
     * @return a string representation of the {@link TabularReport}.
     */
    public String print() {
```

File: plugins/org.eclipse.sirius.ui.debug/src/org/eclipse/sirius/ui/debug/SiriusDebugView.java [hunk 7]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
Comment (lredor) [inline plugins/org.eclipse.sirius.ui.debug/src/org/eclipse/sirius/ui/debug/SiriusDebugView.java:1]: In the class, access to `selection` must be replaced by the call to `getSelection()`. Warnings: There are some method parameters named `selection`, they must not be renamed.

**Review by lredor — CHANGES_REQUESTED** [review]

File: plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java [hunk 8]
```diff
@@ -77,7 +74,7 @@ public Action(String label, String description, Runnable body) {
     /**
      * The currently selection object.
      */
-    protected Object selection;
+    public Object selection;
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java:77]: done

File: packaging/org.eclipse.sirius.debug.feature/pom.xml [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.eclipse.sirius</groupId>
+    <artifactId>sirius-parent</artifactId>
+    <version>7.4.5-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply packaging/org.eclipse.sirius.debug.feature/pom.xml:22]: done

File: packaging/org.eclipse.sirius.debug.feature/feature.xml [hunk 10]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<feature
+      id="org.eclipse.sirius.debug.feature"
+      label="%featureName"
+      version="7.4.5.qualifier"
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply packaging/org.eclipse.sirius.debug.feature/feature.xml:5]: done

File: plugins/org.eclipse.sirius.debug/pom.xml [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+  <parent>
+    <groupId>org.eclipse.sirius</groupId>
+    <artifactId>sirius-parent</artifactId>
+    <version>7.4.5-SNAPSHOT</version>
+    <relativePath>../../packaging/org.eclipse.sirius.parent</relativePath>
+  </parent>
+
+  <artifactId>org.eclipse.sirius.debug</artifactId>
+  <version>7.4.5-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/pom.xml:25]: done

File: plugins/org.eclipse.sirius.debug/META-INF/MANIFEST.MF [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+Bundle-Name: %pluginName
+Bundle-SymbolicName: org.eclipse.sirius.debug;singleton:=true
+Bundle-Version: 7.4.5.qualifier
+Bundle-RequiredExecutionEnvironment: JavaSE-17
+Bundle-Localization: plugin
+Bundle-ActivationPolicy: lazy
+Eclipse-LazyStart: true
+Bundle-Vendor: %providerName
+Export-Package: org.eclipse.sirius.api.debug
+Automatic-Module-Name: org.eclipse.sirius.ui.debug
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/META-INF/MANIFEST.MF:12]: done

File: plugins/org.eclipse.sirius.debug/build.properties [hunk 13]
```diff
@@ -0,0 +1,6 @@
+source.. = src/
+output.. = bin/
+bin.includes = META-INF/,\
+               .,\
+               plugin.properties,\
+               about.html
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/build.properties:6]: done

File: plugins/org.eclipse.sirius.debug/build.properties [hunk 14]
```diff
@@ -0,0 +1,6 @@
+source.. = src/
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/build.properties:1]: done

File: plugins/org.eclipse.sirius.debug/about.html [hunk 15]
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/about.html:1]: done

File: plugins/org.eclipse.sirius.debug/.project [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.SchemaBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.pde.PluginNature</nature>
+		<nature>org.eclipse.jdt.core.javanature</nature>
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/.project:26]: done

File: plugins/org.eclipse.sirius.debug/.project [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+		<buildCommand>
+			<name>org.eclipse.pde.ManifestBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.SchemaBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/.project:22]: done

File: plugins/org.eclipse.sirius.debug/.project [hunk 18]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/.project:1]: done

File: packaging/org.eclipse.sirius.debug.feature/feature.xml [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+      provider-name="%providerName"
+      license-feature="org.eclipse.license"
+      license-feature-version="0.0.0">
+
+   <description url="http://www.eclipse.org/sirius">
+      Sirius Debuging View API.
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply packaging/org.eclipse.sirius.debug.feature/feature.xml:11]: done

File: packaging/org.eclipse.sirius.debug.feature/feature.xml [hunk 20]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<feature
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply packaging/org.eclipse.sirius.debug.feature/feature.xml:2]: done

File: packaging/org.eclipse.sirius.debug.feature/build.properties [hunk 21]
```diff
@@ -0,0 +1,2 @@
+bin.includes = feature.xml,\
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply packaging/org.eclipse.sirius.debug.feature/build.properties:1]: done

File: plugins/org.eclipse.sirius.ui.debug/src/org/eclipse/sirius/ui/debug/SiriusDebugView.java (hunk 7 above)
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.ui.debug/src/org/eclipse/sirius/ui/debug/SiriusDebugView.java:1]: done

File: plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/TabularReport.java (hunk 6 above)
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/TabularReport.java:44]: done

File: plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/TabularReport.java (hunk 5 above)
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/TabularReport.java:40]: done

File: plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java (hunk 3 above)
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java:80]: done

File: plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java (hunk 1 above)
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java:139]: done

File: plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java [hunk 22]
```diff
@@ -135,8 +125,8 @@ public void setFocus() {
      */
     @Override
     public void selectionChanged(IWorkbenchPart part, ISelection selection) {
-        Option<Object> selected = getSelectedElement(selection);
-        if (selected.some()) {
+        Optional<Object> selected = getSelectedElement(selection);
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java:128]: done

File: plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java [hunk 23]
```diff
@@ -180,7 +170,7 @@ protected AdapterFactory getAdapterFactory() {
      * @param text
      *            the text to show in the view's text area.
      */
-    protected void setText(String text) {
+    public void setText(String text) {
         info.setText(text);
     }
```
[the remark this answers is not in the document]
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/src/org/eclipse/sirius/api/debug/AbstractDebugView.java:176]: done

File: plugins/org.eclipse.sirius.debug/.project (hunk 18 above)
[the remark this answers is not in the document]
Comment (lredor) [reply plugins/org.eclipse.sirius.debug/.project:1]: No, the ".checkstyle" is not here.

**Review by lredor — CHANGES_REQUESTED** [review]
The ".checkstyle" file is awlays missing. And there are maybe Copyright to review according to the current specific project rules.

File: plugins/org.eclipse.sirius.debug/.project (hunk 18 above)
[the remark this answers is not in the document]
Comment (florianbarbin) [reply plugins/org.eclipse.sirius.debug/.project:1]: done

**Review by lredor — APPROVED** [review]

**Pull request merged**